connection.go

```diff
@@ -578,6 +577,8 @@ func (me *Connection) openChannel() (*Channel, error) {
 // this connection.
 func (me *Connection) closeChannel(ch *Channel, e *Error) {
 	ch.shutdown(e)
+	me.m.Lock()
+	defer me.m.Unlock()
```

Comment:
Wouldn't the intent be clearer if we explicitly unlocked after `releaseChannel`? or are we after a particular failure that should be guarded against with defer?